server/sonar-server/src/main/java/org/sonar/server/plugins/UpdateCenterClient.java

```diff
@@ -72,10 +72,10 @@
   private UpdateCenter pluginCenter = null;
   private long lastRefreshDate = 0;
 
-  public UpdateCenterClient(UriReader uriReader, Settings settings) throws URISyntaxException {
+  public UpdateCenterClient(UriReader uriReader, Configuration config) throws URISyntaxException {
     this.uriReader = uriReader;
-    this.uri = new URI(settings.getString(URL_PROPERTY));
-    this.isActivated = settings.getBoolean(WebConstants.SONAR_UPDATECENTER_ACTIVATE);
+    this.uri = new URI(config.get(URL_PROPERTY).get());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)